packages/eslint-plugin/tests/lib/rules/no-useless-constructor.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-useless-constructor'),
+  RuleTester = require('eslint').RuleTester;
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module'
+  },
+  parser: '@typescript-eslint/parser'
+});
+
+const error = { message: 'Useless constructor.', type: 'MethodDefinition' };
+
+ruleTester.run('no-useless-constructor', rule, {
+  valid: [
+    'class A { }',
+    'class A { constructor(){ doSomething(); } }',
+    'class A extends B { constructor(){} }',
+    "class A extends B { constructor(){ super('foo'); } }",
+    'class A extends B { constructor(foo, bar){ super(foo, bar, 1); } }',
+    'class A extends B { constructor(){ super(); doSomething(); } }',
+    'class A extends B { constructor(...args){ super(...args); doSomething(); } }',
+    'class A { dummyMethod(){ doSomething(); } }',
+    'class A extends B.C { constructor() { super(foo); } }',
+    'class A extends B.C { constructor([a, b, c]) { super(...arguments); } }',
+    'class A extends B.C { constructor(a = f()) { super(...arguments); } }',
+    'class A extends B { constructor(a, b, c) { super(a, b); } }',
+    'class A extends B { constructor(foo, bar){ super(foo); } }',
+    'class A extends B { constructor(test) { super(); } }',
+    'class A extends B { constructor() { foo; } }',
+    'class A extends B { constructor(foo, bar) { super(bar); } }',
+    // https://github.com/typescript-eslint/typescript-eslint/issues/15
+    'declare class Foo { constructor(); }',
+    'class Foo { constructor(); }',
+    'abstract class Foo { constructor(); }',
+    'abstract class Foo { abstract constructor(); }',
+    // https://github.com/typescript-eslint/typescript-eslint/issues/48
+    'class Foo { constructor(private name: string) {} }',
+    'class Foo { constructor(public name: string) {} }',
+    'class Foo { constructor(protected name: string) {} }'
```

Comment:
I'd like to see a test for the constructor with visibility.

```ts
class Foo { private constructor() {} } // valid!
```

Reply:
i didn't think about accessibility, i will add it